MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/DataProcessorUI/DataProcessorTreeManager.h

```diff
@@ -0,0 +1,108 @@
+#ifndef MANTIDQTMANTIDWIDGETS_DATAPROCESSORTREEMANAGER_H
+#define MANTIDQTMANTIDWIDGETS_DATAPROCESSORTREEMANAGER_H
+
+#include "MantidAPI/ITableWorkspace_fwd.h"
+#include <memory>
+#include <vector>
+#include <map>
+
+class QAbstractItemModel;
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+class DataProcessorCommand;
+class DataProcessorWhiteList;
+
+/** @class DataProcessorTreeManager
+
+DataProcessorTreeManager is an abstract base class defining some methods meant
+to be used by the Generic Data Processor presenter, which will delegate some
+functionality to concrete tree manager implementations, depending on whether or
+not a post-processing algorithm has been defined.
+
+Copyright &copy; 2011-16 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>.
+Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+
+class DataProcessorTreeManager {
```

Comment:
Hmm looking at the public methods it looks more like a table to me (even it it's using a tree internally). `deleteRow`, `appendRow` etc.

`DataProcessorTable` ? Maybe you can think of something better.